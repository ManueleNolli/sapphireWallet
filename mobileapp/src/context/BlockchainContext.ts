import { createContext } from 'react'
import { NETWORKS } from '../constants/Networks'

type BlockchainContextType = {
  currentNetwork: NETWORKS
  setCurrentNetwork: (network: NETWORKS) => void
  ethersProvider: any
  setEthersProvider: (network: NETWORKS) => Promise<void>
}

export const BlockchainContext = createContext<BlockchainContextType>(<
  BlockchainContextType
>{
  currentNetwork: NETWORKS.LOCALHOST,
  setCurrentNetwork: (network: NETWORKS): void => {},
  ethersProvider: null,
  setEthersProvider: async (network: NETWORKS): Promise<void> => {},
})
