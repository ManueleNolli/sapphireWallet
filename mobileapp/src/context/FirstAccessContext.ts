import { createContext } from 'react'

type FirstAccessContextType = {
  isFirstAccess: boolean
  toggleFirstAccess: () => Promise<void>
}

export const FirstAccessContext = createContext<FirstAccessContextType>({
  isFirstAccess: true,
  toggleFirstAccess: async (): Promise<void> => {},
})
