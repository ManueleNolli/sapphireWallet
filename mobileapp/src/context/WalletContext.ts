import { createContext } from 'react'

type WalletContextType = {
  setPrivateKey: (privateKey: string) => Promise<void>
  getPrivateKey: (purposeMessage: string) => Promise<string>
  getEOAAddress: () => string
  setEOAAddress: (eoaAddress: string) => Promise<void>
  getWalletContractAddress: () => string
  setWalletContractAddress: (walletContractAddress: string) => Promise<void>
  resetWallet: () => Promise<void>
}

export const WalletContext = createContext<WalletContextType>(<
  WalletContextType
>{
  setPrivateKey: async (privateKey: string): Promise<void> => {},
  getPrivateKey: async (purposeMessage: string): Promise<string> => {
    console.warn('WalletContext.getPrivateKey not implemented')
    return Promise.resolve('')
  },
  getEOAAddress: (): string => {
    console.warn('WalletContext.getEOAAddress not implemented')
    return ''
  },
  setEOAAddress: async (eoaAddress: string): Promise<void> => {},
  getWalletContractAddress: (): string => {
    console.warn('WalletContext.getWalletContractAddress not implemented')
    return ''
  },
  setWalletContractAddress: async (
    walletContractAddress: string
  ): Promise<void> => {},
  resetWallet: async (): Promise<void> => {},
})
