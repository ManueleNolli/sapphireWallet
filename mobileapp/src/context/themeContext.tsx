import { Context, createContext } from 'react'

type ThemeContextType = {
  theme: string
  toggleTheme: () => void
}

export const ThemeContext: Context<ThemeContextType> = createContext({
  theme: 'light',
  toggleTheme: () => {},
})
