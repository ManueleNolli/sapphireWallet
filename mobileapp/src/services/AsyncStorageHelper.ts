import AsyncStorage from '@react-native-async-storage/async-storage'
export const storeData = async (key: string, value: string): Promise<void> => {
  await AsyncStorage.setItem(key, value).catch((e) => {
    console.error('Error storing data', e)
  })
}

export const getData = async (key: string): Promise<string | null> => {
  const value = await AsyncStorage.getItem(key).catch((e) => {
    console.error('Error getting data', e)
  })
  if (value !== null && value !== undefined) {
    return value
  }
  return null
}
