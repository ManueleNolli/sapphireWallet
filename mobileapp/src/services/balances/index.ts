export * from './balanceCrypto'
export * from './balanceNFTs'
