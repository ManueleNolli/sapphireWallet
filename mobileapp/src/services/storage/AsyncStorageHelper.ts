import AsyncStorage from '@react-native-async-storage/async-storage'

/**
 * Store data in AsyncStorage
 * @param key
 * @param value
 */
export const storeData = async (key: string, value: string): Promise<void> => {
  await AsyncStorage.setItem(key, value).catch((e) => {
    console.error('Error storing data', e)
  })
}

/**
 * Get data from AsyncStorage
 * @param key
 */
export const getData = async (key: string): Promise<string | null> => {
  const value = await AsyncStorage.getItem(key).catch((e) => {
    console.error('Error getting data', e)
  })
  if (value !== null && value !== undefined) {
    return value
  }
  return null
}

/**
 * Remove data from AsyncStorage
 * @param key
 */
export const removeData = async (key: string): Promise<void> => {
  await AsyncStorage.removeItem(key).catch((e) => {
    console.error('Error removing data', e)
  })
}


