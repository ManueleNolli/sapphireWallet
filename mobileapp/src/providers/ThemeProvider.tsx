import React, { useEffect, useState } from 'react'
import { ThemeContext } from '../context/ThemeContext'
import * as eva from '@eva-design/eva'
import { default as theme } from '../../assets/theme/theme.json'
import { ApplicationProvider } from '@ui-kitten/components'
import { getData, storeData } from '../services/storage/'
import constants from '../constants/Constants'
import useLoading from '../hooks/useLoading'

import Loading from '../pages/Loading/Loading'

type ThemeProviderProps = {
  children: React.ReactNode
}

export function ThemeProvider({ children }: ThemeProviderProps) {
  const { isLoading, setIsLoading } = useLoading(true)
  const [themeColor, setThemeColor] = useState('light')

  useEffect(() => {
    const initializeTheme = async () => {
      const theme = await getData(constants.asyncStoreKeys.theme)
      if (!theme) {
        await storeData(constants.asyncStoreKeys.theme, 'light')
        setThemeColor('light')
      } else {
        setThemeColor(theme)
      }
    }

    initializeTheme().then(() => setIsLoading(false))
  }, [])

  const toggleTheme = async () => {
    const nextTheme = themeColor === 'light' ? 'dark' : 'light'
    await storeData(constants.asyncStoreKeys.theme, nextTheme)
    setThemeColor(nextTheme)
  }

  return (
    <ThemeContext.Provider value={{ theme: themeColor, toggleTheme }}>
      <ApplicationProvider
        {...eva}
        theme={{
          ...(themeColor === 'light' ? eva.light : eva.dark),
          ...theme,
        }}
      >
        {isLoading ? <Loading text={'Checking preferences...'} /> : children}
      </ApplicationProvider>
    </ThemeContext.Provider>
  )
}
