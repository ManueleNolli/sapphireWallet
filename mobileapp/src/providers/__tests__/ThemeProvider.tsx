import React from 'react'
import { Button, Text } from 'react-native'
import { getData } from '../../services/AsyncStorageHelper'
import { act, fireEvent, waitFor } from '@testing-library/react-native'
import renderWithTheme from '../../TestHelper'
import { ThemeProvider } from '../ThemeProvider'
import { ThemeContext } from '../../context/ThemeContext'

// MOCKS
const MockComponent = () => {
  return <Text>Mock</Text>
}

jest.mock('../../pages/Loading/Loading')
jest.mock('../../services/AsyncStorageHelper')
describe('ThemeProvider', () => {
  it('Load Children', async () => {
    ;(getData as jest.Mock).mockReturnValue(Promise.resolve('light'))

    let tree

    await waitFor(async () => {
      tree = renderWithTheme(
        <ThemeProvider>
          <MockComponent />
        </ThemeProvider>
      )
    })

    expect(tree).toMatchSnapshot()
  })

  it('Loading', async () => {
    ;(getData as jest.Mock).mockReturnValue(Promise.resolve('light'))

    await waitFor(() => {
      let tree = renderWithTheme(
        <ThemeProvider>
          <MockComponent />
        </ThemeProvider>
      )
      expect(tree).toMatchSnapshot()
      Promise.resolve()
    })
  })

  it('First Access', async () => {
    ;(getData as jest.Mock).mockReturnValue(Promise.resolve(null))

    const MockTheme = () => {
      const { theme } = React.useContext(ThemeContext)
      return (
        <div>
          <Text>{theme}</Text>
        </div>
      )
    }

    let tree
    await waitFor(() => {
      tree = renderWithTheme(
        <ThemeProvider>
          <MockTheme />
        </ThemeProvider>
      )
    })

    expect(tree!.getByText('light')).toBeTruthy()
  })

  it('Toggle theme', async () => {
    ;(getData as jest.Mock).mockReturnValue(Promise.resolve('dark'))

    const MockComponentToggle = () => {
      const { toggleTheme, theme } = React.useContext(ThemeContext)
      return (
        <div>
          <Text>{theme}</Text>
          <Button onPress={() => toggleTheme()} title={'Toggle'} />
        </div>
      )
    }

    let tree

    await waitFor(async () => {
      tree = renderWithTheme(
        <ThemeProvider>
          <MockComponentToggle />
        </ThemeProvider>
      )
    })
    expect(tree).toMatchSnapshot()
    expect(tree!.getByText('dark')).toBeTruthy()

    await act(async () => {
      const button = tree!.getByRole('button')
      fireEvent.press(button)
    })

    expect(tree).toMatchSnapshot()
    expect(tree!.getByText('light')).toBeTruthy()
  })
})
