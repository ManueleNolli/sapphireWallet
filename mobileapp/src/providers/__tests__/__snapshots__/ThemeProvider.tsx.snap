// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeProvider Load Children 1`] = `
<Text>
  Mock
</Text>
`;

exports[`ThemeProvider Loading 1`] = `
<div
  data-testid="Loading"
>
  Loading
</div>
`;

exports[`ThemeProvider Toggle theme 1`] = `
<div>
  <Text>
    dark
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Toggle
      </Text>
    </View>
  </View>
</div>
`;

exports[`ThemeProvider Toggle theme 2`] = `
<div>
  <Text>
    light
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Toggle
      </Text>
    </View>
  </View>
</div>
`;
